TopsortAnalytics/src/main/java/com/topsort/analytics/banners/BannerConfig.kt

```diff
@@ -0,0 +1,38 @@
+package com.topsort.analytics.banners
+
+sealed class BannerConfig private constructor() {
+    data class LandingPage(
+        val slotId: String,
+        val ids: List<String>,
+        val device: String? = null,
+        val geoTargeting: String? = null
+    ) : BannerConfig()
+
+    data class CategorySingle(
+        val slotId: String,
+        val category: String,
+        val device: String? = null,
+        val geoTargeting: String? = null
+    ) : BannerConfig()
+
+    data class CategoryMultiple(
+        val slotId: String,
+        val categories: List<String>,
+        val device: String? = null,
+        val geoTargeting: String? = null,
+    ) : BannerConfig()
+
+    data class CategoryDisjuntions(
+        val slotId: String,
+        val disjunctions: List<List<String>>,
+        val device: String? = null,
+        val geoTargeting: String? = null,
+    ) : BannerConfig()
+
+    data class Keyword(
+        val slotId: String,
+        val keyword: String,
+        val device: String? = null,
+        val geoTargeting: String? = null,
+    ) : BannerConfig()
+}
```

Comment:
nit: Newline at the end of files 
If using Android studio: Settings > Editor > General > On save [Ensure every saved file ends with a line break]

Reply:
Can we have these configs in a project level file that Android Studio can pick up?

Reply:
You can checkout the .idea folder into git